hil/src/ssh_wrapper.rs

```diff
@@ -0,0 +1,127 @@
+use color_eyre::{eyre::bail, Result};
+use ssh2::Session as Ssh2Session;
```

Comment:
instead of ssh2, which requires you to dynamically link against C libraries, why not just use https://docs.rs/cmd_lib/latest/cmd_lib/ (blocking but terse) or https://docs.rs/tokio/latest/tokio/process/struct.Command.html (async but verbose) and the actual `ssh` CLI.

Alternatively why not https://github.com/Eugeny/russh which actually does async in native rust